utils/js/VersionUtils.js

```diff
@@ -19,17 +19,29 @@ export function checkVersion(version, minVersion, maxVersion) {
  * @return boolean
  */
 export function checkMinVersion(version, minVersion) {
-    // Treat snapshot versions as the release version
-    version = version.replace('-SNAPSHOT', '.0');
-    return version && minVersion && semver.satisfies(version, '>=' + minVersion);
+    return (
+        version && minVersion &&
+        semver.satisfies(normalizeVersion(version), '>=' + normalizeVersion(minVersion))
+    );
 }
 
 /**
  * Check if a version string meets a maximum version
  * @return boolean
  */
 export function checkMaxVersion(version, maxVersion) {
-    // Treat snapshot versions as the release version
-    version = version.replace('-SNAPSHOT', '.0');
-    return version && maxVersion && semver.satisfies(version, '<=' + maxVersion);
+    return (
+        version && maxVersion &&
+        semver.satisfies(normalizeVersion(version), '<=' + normalizeVersion(maxVersion))
+    );
+}
+
+/**
+ * Normalizes a Java/Maven style x.0-SNAPSHOT version to a semver compatible x.0.0-SNAPSHOT string.
+ * @param {string} version
+ * @return {string} - normalized version, if input matched as above, or input version unmodified.
+ */
+export function normalizeVersion(version) {
+    const isTwoDigitSnap = /^\d+\.0-SNAPSHOT$/.test(version);
```

Comment:
nice!

Reply:
If we are going to export this as `normalizeVersion` one wonders why we wouldn't be normalizing XH.appVersion and XH.clientVersion and the other versions in environmentService.  Minor nit -- its fine, I think